### ISSUE TITLE
Split `runtime/interpreter/value.go` into multiple files

### ISSUE THREAD

**Issue by darkdrag00nv2** [onflow/cadence#2566]
### Issue to be solved

The `runtime/interpreter/value.go` file is huge. Even the GitHub syntax highlighting gets disabled on PR reviews because of the huge size.

### Suggested Solution

We should consider splitting the file into different files. A decent split can be:

1. `value_collections.go`
2. `value_numbers.go`
3. `value_options.go`
.. More if needed ..

`value_test.go` would have to be split accordingly.

**Comment by turbolent** [comment]
Agreed, it's grown significantly and new value types have already been added in new files.

Logistically, the plan is to merge `feature/stable-cadence` into `master` soon, when all maybe features have been merged (e.g. Entitlements). Once that is done, and before other major PRs are started which might touch `value.go`, we should split it up 👍
